lib/transbank/sdk/patpass/patpass_by_webpay/transaction.rb

```diff
@@ -0,0 +1,73 @@
+module Transbank
+  module Patpass
+    module PatpassByWebpay
+      class Transaction
+        extend Transbank::Utils::NetHelper
+        CREATE_TRANSACTION_ENDPOINT = 'rswebpaytransaction/api/webpay/v1.0/transactions'
+        COMMIT_TRANSACTION_ENDPOINT = 'rswebpaytransaction/api/webpay/v1.0/transactions'
+        TRANSACTION_STATUS_ENDPOINT = 'rswebpaytransaction/api/webpay/v1.0/transactions'
+
+        class << self
+
+          def create(buy_order:, session_id:, amount:, return_url:, options: nil)
+
+            api_key = options.api_key || default_integration_params[:api_key]
+            commerce_code = options.commerce_code || default_integration_params[:api_key]
+            base_url = PatpassByWebpay::Base.integration_types[options.integration_type] || default_integration_params[:base_url]
+
+            body = {
+                buy_order: buy_order, session_id: session_id,
+                amount: amount, return_url: return_url
+            }
+
+            url = base_url + CREATE_TRANSACTION_ENDPOINT
+            headers = webpay_headers(commerce_code: commerce_code, api_key: api_key)
+
+            resp = http_post(uri_string: url, body: body, headers: headers, camel_case_keys: false)
+            body = JSON.parse(resp.body)
+
+            return ::Transbank::Patpass::PatpassByWebpay::TransactionCreateResponse.new(body) if resp.kind_of? Net::HTTPSuccess
+            raise Errors::TransactionCreateError.new(body['error_message'], resp.code)
+          end
+
+          def commit(token:, options: nil)
+
+            api_key = options.api_key || default_integration_params[:api_key]
+            commerce_code = options.commerce_code || default_integration_params[:api_key]
+            base_url = PatpassByWebpay::Base.integration_types[options.integration_type] || default_integration_params[:base_url]
+
+            url = base_url + COMMIT_TRANSACTION_ENDPOINT + "/#{token}"
+            headers = webpay_headers(commerce_code: commerce_code, api_key: api_key)
+
+            resp = http_put(uri_string: url, body: nil, headers: headers)
+            body = JSON.parse(resp.body)
+            return ::Transbank::Patpass::PatpassByWebpay::TransactionCommitResponse.new(body) if resp.kind_of? Net::HTTPSuccess
+            raise Errors::TransactionCommitError.new(body['error_message'], resp.code)
+          end
+
+          def status(token:, options: nil)
+            api_key = options.api_key || default_integration_params[:api_key]
+            commerce_code = options.commerce_code || default_integration_params[:api_key]
+            base_url = WebpayPlus::Base.integration_types[options.integration_type] || default_integration_params[:base_url]
```

Comment:
I think this call is wrong, not necessarily in whether it works or not, but conceptually.
Patpass is a different product family from WebpayPlus

Reply:
@eugenioLopezRamos is right, in this case you should use a Patpass by webpay base, because the user can use different product as the same time with different configuration so we can use `WebpayPlus::Base` on patpass, onepay or Transacción Completa.